baselinemarkers/get_marker_genes_proteomics.R

```diff
@@ -0,0 +1,466 @@
+
+
+
+# add column with gene_name, join on gene ID
+final_df <- final_df %>%
+    left_join(expression_decorated %>% select(GENE_ID = Gene.ID, GENE_NAME = Gene.Name),
+              by = "GENE_ID")
+
+
+
+# Ensure EXPRESSION is numeric
+final_df$EXPRESSION <- NA_real_
+
+# Loop over rows to update EXPRESSION based on matching GENE_ID and GROUP
+for (i in seq_len(nrow(final_df)) ) {
+    gene_id <- final_df$GENE_ID[i]
+    group <- final_df$GROUP_NAME[i]
+    
+    # Check if gene exists in expression_decorated and group column exists
+    row_match <- which(expression_decorated$Gene.ID == gene_id)
+
+    if (length(row_match) == 1 && group %in% colnames(expression_decorated)) {
+
+        expr_values <- as.numeric(expression_decorated[row_match, group]) # as.numeric(unlist(strsplit(expr_string, ",")))
+
+        final_df$EXPRESSION[i] <- expr_values
+    }
+}
+
+
+
+###############################################################
+# apply cutoff of gene expression and reorder ranking
+###############################################################
+
+# Identify genes where any row has RANKING != -1 AND EXPRESSION <= EXPRESSION_CUTOFF
+genes_to_remove <- final_df %>%
+    filter(RANKING != -1, EXPRESSION <= EXPRESSION_CUTOFF) %>%
+    distinct(GENE_ID) %>%
+    pull(GENE_ID)
+
+# remove all rows for those genes
+filtered_df <- final_df %>%
+    filter(!GENE_ID %in% genes_to_remove)
+
+# reassign RANKING within each GROUP_NAME for remaining valid rows
+# Re-rank genes within each group by ascending specificity, then descending expression, then gene ID (if RANKING != -1)
+
+filtered_df <- filtered_df %>%
+  group_by(GROUP_NAME) %>%
+  arrange(SPECIFICITY_SCORE, desc(EXPRESSION), GENE_ID, .by_group = TRUE) %>%
+  mutate(
+    new_rank = NA_integer_,  # initialize
+    new_rank = replace(new_rank, RANKING != -1, row_number()),  # assign ranks where applicable
+    RANKING = if_else(is.na(new_rank), -1L, new_rank)
+  ) %>%
+  select(-new_rank) %>%
+  ungroup()
+
+
+
+# get unique genes ordered by RANKING within each group using reframe()
+ordered_genes <- filtered_df %>%
+    filter(RANKING > 0) %>%
+    group_by(GROUP_NAME) %>%
+    arrange(RANKING, .by_group = TRUE) %>%
+    reframe(GENE_ID = unique(GENE_ID)) %>%
+    pull(GENE_ID)
+
+
+# Use those ordered genes to sort the full dataframe
+filtered_df_sorted <- filtered_df %>%
+    filter(GENE_ID %in% ordered_genes) %>%
+    mutate(GENE_ID = factor(GENE_ID, levels = ordered_genes)) %>%
+    arrange(GENE_ID, GROUP_NAME)
+
+
+# replace -1 with NULL for sql loading
+filtered_df_sorted <- filtered_df_sorted %>%
+    mutate(
+        RANKING = ifelse(RANKING == -1, "NULL", as.character(RANKING)),
+        SPECIFICITY_SCORE = ifelse(SPECIFICITY_SCORE == -1, "NULL", as.character(SPECIFICITY_SCORE))
+    )
+
+
+colnames(filtered_df_sorted) <- c("experiment_accession", "assay_id", "assay", "gene_id", "specificity_score", "marker_gene_rank", "expression_unit", "number_assays", "expression_level", "gene_name")
```

Comment:
```suggestion
colnames(filtered_df_sorted) <- c("experiment_accession", "assay", "gene_id", "specificity_score", "marker_gene_rank", "expression_unit", "number_assays", "expression_level", "gene_name", "assay_id")
```

Moving `assay_id` to end to match schema